### PR TITLE
List utility

### PR DESCRIPTION
**Pull request by adamhutchings**

### CODE REVIEW

**Review by JakeRoggenbuck — COMMENTED** [review]
Looks good. Assuming you're going to add lset and ladd in this PR

**Comment by adamhutchings** [comment]
Correct, but we don't need this list implementation yet so those can wait.

**Comment by TheCrystalKeeper** [comment]
Looks good.

File: src/util/test_list.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#include "test_list.h"
+#include <testing/tassert.h>
+#include <testing/test_utils.h>
+
+// A particular application I want to test out.
+int basic_100_test() {
```
Comment (JakeRoggenbuck) [inline src/util/test_list.c:6]: Good test

**Review by JakeRoggenbuck — APPROVED** [review]

File: src/util/list.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        // Make a new block and put the element in there.
+        l->tail = new_block(l);
+        lb->next = l->tail;
+        lb = lb->next;
+        lb->array[lb->full++] = element;
+    }
+    return 0;
+}
+
+int literate(List* l, int (*fn)(void*)) {
```
Comment (JakeRoggenbuck) [inline src/util/list.c:127]: This is cool

**Pull request merged**